File: cpp/leap/leap.cpp
#include "leap.h"

bool divisible_by(int n, int year) {
  return year % n == 0;
}

bool leap::is_leap_year(int year) {
  return divisible_by(400, year) ||
         (divisible_by(4, year) && !divisible_by(100, year));
};

File: csharp/leap/Leap.cs
using System;

public static class IntExtensions
{
    public static bool DivisibleBy(this int x, int factor)
    {
        return x % factor == 0;
    }
}

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        return year.DivisibleBy(400) ||
               (year.DivisibleBy(4) && !year.DivisibleBy(100));
    }
}
